### DISCUSSION TITLE
MitmProxy certification issue to allow https traffic on local system.

### DISCUSSION THREAD

**Discussion by anupsah1** [mitmproxy/mitmproxy#6779] [General]
Hi, I have setup an ubuntu ec2 instance and deploy the mitmproxy official docker image and created the container. But when i am using this proxy server to my VS CODE it is blocking my Marketplace extensions in VS code and that VS code is deployed to my local machine. Here i the certificates are already present in the ec2 instance. Any one know how to fix this issue. I do not want the certificate to be installed in my local machine.
<img width="906" alt="mitiproxy-certs" src="https://github.com/mitmproxy/mitmproxy/assets/162571160/9fef546b-f95c-4dfd-8534-f895180d5a31">

**Comment by rosydawn6** [comment]
Typically the pem file must be installed on your mitmproxy server but you need to install the corresponding .crt certificate files on the client - your laptop.  If you don't want to install those certs on your laptop you will need to filter the https traffic.  It may be the case the Vs Code marketplace extension infrastructure uses https.

**Reply by anupsah1** [reply]
@rosydawn6 yes Vs Code marketplace extension infrastructure uses https. Is there any way that without providing the .crt to the client server (means my laptop) i can access the https request? And also i have saved .crt file to windows and it is working but for linux and macos is not working. So can you tell me the process for the linux and macos as well.

**Comment by anupsah1** [comment]
yes Vs Code marketplace extension infrastructure uses https. Is there any way that without providing the .crt to the client server (means my laptop) i can access the https request? And also i have saved .crt file to windows and it is working but for linux and macos is not working. So can you tell me the process for the linux and macos as well.

**Reply by rosydawn6** [reply]
There is no way to access the https request without installing the crt file on the client device.

**Comment by anupsah1** [comment]
Thank You for the confirmation @rosydwan6. 
So, can you guide me how to install the certificate on linux and macos. I tried through the document provided by mitmproxy. But it is not working? Can you help me with steps need to be followed?

**Reply by rosydawn6** [reply]
Each system has a bit different procedure. I recommend googling the procedure for your specific system/browser etc...
